hoomd/md/TwoStepRATTLEBD.h

```diff
@@ -0,0 +1,476 @@
+
+    GPUArray<Scalar3> tmp_f_brownian(group_size, m_exec_conf);
+
+    m_f_brownian.swap(tmp_f_brownian);
+
+    ArrayHandle<Scalar3> h_f_brownian(m_f_brownian, access_location::host);
+
+    for (unsigned int i = 0; i < group_size; i++)
+        {
+        h_f_brownian.data[i].x = 0;
+        h_f_brownian.data[i].y = 0;
+        h_f_brownian.data[i].z = 0;
+        }
+    }
+
+/*! \param timestep Current time step
+    \post Particle positions are moved forward to timestep+1
+
+    The integration method here is from the book "The Langevin and Generalised Langevin Approach to the Dynamics of
+    Atomic, Polymeric and Colloidal Systems", chapter 6.
+*/
+
+template< class Manifold>
+void TwoStepRATTLEBD<Manifold>::integrateStepOne(unsigned int timestep)
+    {
+    unsigned int group_size = m_group->getNumMembers();
+
+    const Scalar currentTemp = (*m_T)(timestep);
+
+    // profile this step
+    if (m_prof)
+        m_prof->push("BD step 1");
+
+    const GlobalArray< Scalar4 >& net_force = m_pdata->getNetForce();
+    ArrayHandle<Scalar4> h_vel(m_pdata->getVelocities(), access_location::host, access_mode::readwrite);
+    ArrayHandle<Scalar4> h_pos(m_pdata->getPositions(), access_location::host, access_mode::readwrite);
+    ArrayHandle<int3> h_image(m_pdata->getImages(), access_location::host, access_mode::readwrite);
+    ArrayHandle<unsigned int> h_rtag(m_pdata->getRTags(), access_location::host, access_mode::read);
+
+    ArrayHandle<Scalar4> h_net_force(net_force, access_location::host, access_mode::read);
+    ArrayHandle<Scalar3> h_f_brownian(m_f_brownian, access_location::host, access_mode::read);
+    ArrayHandle<Scalar> h_gamma(m_gamma, access_location::host, access_mode::read);
+    ArrayHandle<Scalar> h_diameter(m_pdata->getDiameters(), access_location::host, access_mode::read);
+
+    ArrayHandle<Scalar3> h_gamma_r(m_gamma_r, access_location::host, access_mode::read);
+    ArrayHandle<Scalar4> h_orientation(m_pdata->getOrientationArray(), access_location::host, access_mode::readwrite);
+    ArrayHandle<Scalar4> h_torque(m_pdata->getNetTorqueArray(), access_location::host, access_mode::readwrite);
+
+    ArrayHandle<Scalar4> h_angmom(m_pdata->getAngularMomentumArray(), access_location::host, access_mode::readwrite);
+    ArrayHandle<Scalar3> h_inertia(m_pdata->getMomentsOfInertiaArray(), access_location::host, access_mode::read);
+
+    const BoxDim& box = m_pdata->getBox();
+
+    // perform the first half step
+    // r(t+deltaT) = r(t) + (Fc(t) + Fr)*deltaT/gamma
+    // iterative: r(t+deltaT) = r(t+deltaT) - J^(-1)*residual
+    // v(t+deltaT) = random distribution consistent with T
+    for (unsigned int group_idx = 0; group_idx < group_size; group_idx++)
+        {
+        unsigned int ptag = m_group->getMemberTag(group_idx);
+        unsigned int j = h_rtag.data[ptag];
+
+        // Initialize the RNG
+        RandomGenerator rng(RNGIdentifier::TwoStepBD, m_seed, ptag, timestep);
```

Comment:
```suggestion
        RandomGenerator rng(RNGIdentifier::TwoStepBD, m_seed, ptag, timestep, 1);
```
Add a unique counter value so that this generates RNG numbers unique from those in `includeRATTLEForce`.